src/transport/stomp/stomp.client.ts

```diff
@@ -147,8 +146,8 @@ export default class StompClient {
     headers: Record<string, string> = {},
     body = '',
   ) {
-    while (!this.opened || !this.connected) {
-      await sleep(1000);
+    if (!this.opened || !this.connected) {
+      await this.reconnect();
```

Comment:
Can we get here on start?
When socket is not opened yet?

Reply:
At the start, we use the `connect` method. Here we are checking to connect status and reconnecting if it needs. And here we are handling parallel reconnecting using `reconnectionPromise`